ReactAndroid/src/main/java/com/facebook/react/uimanager/UIImplementation.java

```diff
@@ -279,23 +285,25 @@ public void profileNextBatch() {
    * Invoked by React to create a new node with a given tag, class name and properties.
    */
   public void createView(int tag, String className, int rootViewTag, ReadableMap props) {
-    ReactShadowNode cssNode = createShadowNode(className);
-    ReactShadowNode rootNode = mShadowNodeRegistry.getNode(rootViewTag);
-    Assertions.assertNotNull(rootNode, "Root node with tag " + rootViewTag + " doesn't exist");
-    cssNode.setReactTag(tag);
-    cssNode.setViewClassName(className);
-    cssNode.setRootTag(rootNode.getReactTag());
-    cssNode.setThemedContext(rootNode.getThemedContext());
-
-    mShadowNodeRegistry.addNode(cssNode);
+    synchronized (uiImplementationThreadLock) {
+      ReactShadowNode cssNode = createShadowNode(className);
+      ReactShadowNode rootNode = mShadowNodeRegistry.getNode(rootViewTag);
+      Assertions.assertNotNull(rootNode, "Root node with tag " + rootViewTag + " doesn't exist");
+      cssNode.setReactTag(tag); // Thread safety needed here
+      cssNode.setViewClassName(className);
+      cssNode.setRootTag(rootNode.getReactTag());
+      cssNode.setThemedContext(rootNode.getThemedContext());
+
+      mShadowNodeRegistry.addNode(cssNode);
```

Comment:
I think so yeah, it's a timing issue, that part of the code is super volatile to bugs.
I can't give you a definite answer, but on the other hand I don't think it affects performance, because that call only assigns a value to a property and not much else https://github.com/facebook/react-native/blob/a07b66026da6168f59aafb808cd625ab868a6d00/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNodeImpl.java#L559